### ISSUE TITLE
Bug: google engine

### ISSUE THREAD

**Issue by GitTimeraider** [searxng/searxng#2389]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
Repository: https://github.com/searxng/searxng
Branch: master
Version: 2023.4.25+216751da

**How did you install SearXNG?**
Installed through Docker
**What happened?**
This keeps happening a LOT lately for the google engine. As im the only one using the instance, I doubt there are to many requests coming from me :D

**How To Reproduce**
Usually halfway into an day (maybe 20-30 searches done over the morning) it stops working and gives this error

**Expected behavior**
Hoping my small amount of uses would have been enough to escape any Google limits, but might not be the case.

**Screenshots & Logs**

**Additional context**
Saw some people comment with the fact that they use "use_mobile_ui" or something similar but I dont think thats relevant anymore right? (Seeing as it doesnt use that parameter anymore if I look into the code ) 

Just tell me what data you need if more is needed and ill try to retrieve it with my limited knowledge :P

**Technical report**

Error
  * Error: searx.exceptions.SearxEngineTooManyRequestsException
  * Percentage: 100
  * Parameters: `('Too many request',)`
  * File name: `searx/search/processors/online.py:118`
  * Function: `_send_http_request`
  * Code: `response = req(params['url'], **request_args)`

![2023-04-28 17_18_02-Madnesshell Search - Vivaldi](https://user-images.githubusercontent.com/57343973/235187322-e7996e53-aed4-4c3b-8463-edbba4fa7f74.png)

**Comment by return42** [comment]
> As im the only one using the instance, I doubt there are to many requests coming from me :D
> .. Usually halfway into an day (maybe 20-30 searches done over the morning) it stops working and gives this error

Should not happen .. is your instance public / can I have a look .. you can send me an eMail markus.heiser@darmarit.de if you don't want to post the URL here.

> Saw some people comment with the fact that they use "use_mobile_ui" o

This option does no longer exists, Google-WEB engine should work *out of the box* without blocking.

**Comment by return42** [comment]
Activate the limiter.

> Saw some people comment with the fact that they use "use_mobile_ui" o

This option does no longer exists, Google-WEB engine should work *out of the box* without blocking.

**Comment by unixfox** [comment]
> Activate the limiter.

But he is the only user of the instance like he said so this won't help.

**Comment by unixfox** [comment]
@GitTimeraider what's your server provider? Or is it installed locally on your own personal machine?

**Comment by GitTimeraider** [comment]
> @GitTimeraider what's your server provider? Or is it installed locally on your own personal machine?

It is hosted on a NAS located in my home

**Comment by GitTimeraider** [comment]
> Activate the limiter.
> 
> > Saw some people comment with the fact that they use "use_mobile_ui" o
> 
> This option does no longer exists, Google-WEB engine should work _out of the box_ without blocking.

Hope you dont mind me being sarcastic and applauding you with how you managed to fix a issue by not addressing the actual issue... Or at least, not after editing your comment.

**Comment by GitTimeraider** [comment]
> Activate the limiter.
> 
> > Saw some people comment with the fact that they use "use_mobile_ui" o
> 
> This option does no longer exists, Google-WEB engine should work _out of the box_ without blocking.

Limiter on an instance being used like 40 times a day? ....
Ah well, will try. If not ill just be back :)

**Comment by return42** [comment]
> But he is the only user of the instance like he said so this won't help.

He send me the public URL .. I tested and have seen that the limiter is not activated.

> It is hosted on a NAS located in my home

Now I'm confused .. is the instance available from the internet or not? .. you send me the subdomain, was it just a temporary setup for me to test?

If the instance is not available from the internet there is no need activate the limiter.

**Comment by GitTimeraider** [comment]
> > But he is the only user of the instance like he said so this won't help.
> 
> He send me the public URL .. I tested and have seen that the limiter is not activated.
> 
> > It is hosted on a NAS located in my home
> 
> Now I'm confused .. is the instance available from the internet or not? .. you send me the subdomain, was it just a temporary setup for me to test?
> 
> If the instance is not available from the internet there is no need activate the limiter.

> > But he is the only user of the instance like he said so this won't help.
> 
> He send me the public URL .. I tested and have seen that the limiter is not activated.
> 
> > It is hosted on a NAS located in my home
> 
> Now I'm confused .. is the instance available from the internet or not? .. you send me the subdomain, was it just a temporary setup for me to test?
> 
> If the instance is not available from the internet there is no need activate the limiter.

Set limiter to true now so ill test and see.

Its hosted at my home on my NAS but publicly available from the internet. The domain DNS gets redirected to an IP/port at my home.
The reason for the subdomain is simply due to the fact that I have some other unrelated webapps with other subdomainnames under the same domain.

**Comment by return42** [comment]
> Its hosted at my home on my NAS but publicly available from the internet. 

Ah, OK ..  thanks for clarify .. I tested you instance again .. and yes; you get very fast a "to many request" from google .. long before the limiter starts to block the requests from my client.

I also did some test on !ddg .. but ddg does not have this issue.

What happens when you clone SearXNG on your desktop and start it locally (by `make run`) on your desktop, then repeat `!go foobar` 20 times in a row.

---

Update: the limiter is still not active on your instance .. you can test it by

    curl 'https://search.example.org/search?q=%21go%20foobar'

**Comment by GitTimeraider** [comment]
> 

Weird. Doublechecked by the yml file really shows:
_limiter: true  # rate limit the number of request on the instance, block some bots_
Should be everything I have to change right?

**Comment by kiddailey** [comment]
For what it's worth - I've been seeing this issue a lot lately (and also "Suspended: access denied") while trying various public nodes listed on searx.space.  My initial thought was that maybe there was a bug/API change or something, which is why I went searching here and found this report.

Of course, it could just be random bad luck and I'm happening to hit nodes that are getting a lot of traffic, but it just seemed odd from a timing standpoint.

Was planning on setting my own private node up like the original poster, so I'll be curious to see if this is really an issue.  Especially since my usage might be a bit more due to research usage.

**Comment by return42** [comment]
> For what it's worth - I've been seeing this issue a lot lately (and also "Suspended: access denied") while trying various public nodes listed on searx.space. 

If you have a look at the "Engines" tab at searx.space, in the column "google" you will see that most of the instances return results from the google engine ..

![image](https://user-images.githubusercontent.com/554536/235712570-cd5d3eb3-9166-4edf-99eb-322a7a2c77d2.png)

If there is a  ![image](https://user-images.githubusercontent.com/554536/235713348-92bcf835-a2d8-4906-bf77-cecb450295c8.png) in the google column, its most often a (unmaintained) searx instance (here we are in the SearXNG project).

> Was planning on setting my own private node up like the original poster, so I'll be curious to see if this is really an issue. Especially since my usage might be a bit more due to research usage.

Don't hesitate :+1: / google should work .. we have sometimes this *Too many request* message but this message disappears after suspend-time is ended.

What I mean; it is OK if an instance reports *Too many request* from time to time but what @GitTimeraider reported here is very special to his instance  / and I don't have a clue what the cause is.

**Comment by GitTimeraider** [comment]
Hmm, I thought it was going well for a day or 2 after enabling the limiter in the settings.yml, however this morning it only took me like 5 searches before it went back to giving the to many requests message.

**Comment by return42** [comment]
> however this morning it only took me like 5 searches before it went back to giving the to many requests message.

>> but what @GitTimeraider reported here is very special to his instance / and I don't have a clue what the cause is.

@GitTimeraider, we need to analyse why your instances has this issue .. did you read my post above:

[What happens when you clone SearXNG on your desktop and start it locally (by `make run`) on your desktop, then repeat `!go foobar` 20 times in a row.](https://github.com/searxng/searxng/issues/2389#issuecomment-1529569411)

**Comment by GitTimeraider** [comment]
@return42 Did try it out now (my jobexpertise lies in anti-virus and protection-methods so im not familiar enough with a lot of this stuff to get everything running instantly :D)

Anyway.. installed docker on my desktop and I can do a lot more than 20 searches in a row without it running into any issues. So after that I decided to see how differently my packages went from my desktop compared to my NAS (always nice to have an OPNsense router in this case :P ).

Found something out.. my desktop simply sends and recieves everything on my main internet line. However, while searxng on my NAS does receive and is accessed through my main internet line .. the outgoing connections towards google are not.
Turns out the outgoing connection is going through an VPN I have set up which my torrent and NZB applications use to make their connections. And im quite sure that a general VPN IP does have an tendency to get blocked by google more regularly.

So now ill probably have to struggle a good bit to try and find a way to redirect my outgoing traffic from the NAS through the VPN UNLESS its going towards google. Not sure if I can even manage that ... might end up having to choose whether to either run it without VPN or host my searxng on another device.

Anyhow, all of that is obviously not your problem at all so ill not make it yours anymore. Unless something drastically changes during my further actions, ill chalk it up to this ... 95% sure its the cause.

**Comment by unixfox** [comment]
I'm interested to know the VPN provider so that we can try to find a way to make it work on an IP address that is often blocked.

**Comment by GitTimeraider** [comment]
> I'm interested to know the VPN provider so that we can try to find a way to make it work on an IP address that is often blocked.

@unixfox
Surfshark vpn and im specifically using their Manchester server (have an 24/7 wireguard vpn open towards it)

They do actually have a paid service that gives an dedicated IP so that you get your own wan ip and wont get caught in with other vpn users.
Very tempted to try that out :)

**Comment by GitTimeraider** [comment]
Ok, last comment.

I can fully confirm it was the shared-ip nature of the IP in question.
Even though it is an expensive service .. i was to tempted and bought an dedicated IP for Surfshark VPN (Guess im German now :D ) ... after turning on the vpn connection after adjusting it to the dedicated IP, no issues at all. It works as if im on my local network.
Even let my torrent downloader test an IP torrent and it came back with the VPN IP-address so it has to be working.

Guess thats the whole story now :D
